AISKU/snippet/snippet.js

```diff
@@ -1,106 +1,304 @@
-
-    // Collect global errors
-    // Note: ApplicationInsightsAnalytics is the extension string identifier for
-    //  AppAnalytics. It is defined in ApplicationInsights.ts:ApplicationInsights.identifer
-    if (!(aiConfig.disableExceptionTracking === true ||
-        (aiConfig.extensionConfig &&
-        aiConfig.extensionConfig.ApplicationInsightsAnalytics &&
-        aiConfig.extensionConfig.ApplicationInsightsAnalytics.disableExceptionTracking === true))) {
-
-        method = "onerror";
-        createLazyMethod("_" + method);
-        var originalOnError = localWindow[method];
-        localWindow[method] = function(message, url, lineNumber, columnNumber, error) {
-            var handled = originalOnError && originalOnError(message, url, lineNumber, columnNumber, error);
-            if (handled !== true) {
-                appInsights["_" + method]({
-                    message: message,
-                    url: url,
-                    lineNumber: lineNumber,
-                    columnNumber: columnNumber,
-                    error: error
-                });
+            // apply the default endpoints
+            if (!fields[strIngestionendpoint]) {
+                // use endpoint suffix where overrides are not provided
+                var endpointSuffix = fields.endpointsuffix;
+                // Only fetch the location if a suffix was supplied
+                var fLocation = endpointSuffix ? fields.location : null;
+                fields[strIngestionendpoint] = "https://" + (fLocation ? fLocation + "." : strEmpty) + "dc." + (endpointSuffix || "services.visualstudio.com");
             }
 
-            return handled;
-        };
-        aiConfig.autoExceptionInstrumented = true;
-    }
+            return fields;
+        }
+
+        function _sendEvents(evts, endpointUrl) {
+            if (JSON) {
+                var sender = win.fetch;
+                if (sender && !snipConfig.useXhr) {
+                    sender(endpointUrl, { method:strPostMethod, body: JSON.stringify(evts), mode:"cors"});
+                } else if (XMLHttpRequest) {
+                    // IE doesn't support fetch and private clouds may only be using IE
+                    var xhr = new XMLHttpRequest();
+                    xhr.open(strPostMethod, endpointUrl);
+                    xhr.setRequestHeader("Content-type", "application/json");
+                    xhr.send(JSON.stringify(evts));
+                }
+            }
+        }
+
+        function _reportFailure(targetSrc) {
+            var conString = _parseConnectionString();
+            var iKey = conString[strInstrumentationKey] || aiConfig[strInstrumentationKey] || strEmpty;
+            var ingest = conString[strIngestionendpoint];
+            var endpointUrl = ingest ? ingest + "/v2/track" : config.endpointUrl; // only add /v2/track when from connectionstring
+
+            var message = "SDK LOAD Failure: Failed to load App Insights Sdk script (See stack for details)";
```

Comment:
Updated per Ram's email comments

Reply:
I still see App Insights.  We should call this Application Insights.  and capitialize Sdk?

Reply:
Would also add JavaScript SDK to the message, if it isn't visible anywhere else.

Reply:
We wanted to keep the text data to min so we can minimize on snippet bloat.  Fact that we are listing .js file in the error should be good enough IMO.